src/client/Microsoft.Identity.Client/Platforms/Android/AndroidBroker.cs

```diff
@@ -213,5 +213,27 @@ private static MsalTokenResponse CreateMsalTokenResponseFromResult(string broker
 
             return MsalTokenResponse.CreateFromBrokerResponse(response);
         }
+
+        public async Task<IEnumerable<IAccount>> GetAccountsAsync(string clientID)
+        {
+            Dictionary<string, string> brokerPayload = new Dictionary<string, string>();
+            brokerPayload.Add(BrokerParameter.ClientId, clientID);
+
+            try
+            {
+                await _brokerHelper.InitiateBrokerHandshakeAsync(_activity).ConfigureAwait(false);
+
+                return _brokerHelper.GetBrokerAccountsInAccountManager(brokerPayload);
+            }
+            catch (Exception ex)
+            {
+                _logger.Error("Broker get accounts operation failed to complete. In order to perform brokered authentication on android" +
+                    " you need to ensure that you have installed either Intune Company Portal (Version 5.0.4689.0 or greater) or Microsoft Authenticator (6.2001.0140 or greater).");
+                if (ex is MsalException)
```

Comment:
This piece of error handling is repeated at line 62 in `AcquireTokenUsingBrokerAsync` - please consolidate, e.g. via a private method.